test/test.js

```diff
@@ -480,6 +480,17 @@ describe('filing-cabinet', function() {
       assert.equal(resolved, `${directory}/node_modules/resolve/index.js`);
     });
 
+    it('resolves a path using the resolve root config', function() {
+      const resolved = cabinet({
+        partial: 'ast',
+        filename: `${directory}/test/ast.js`,
+        directory: `${directory}/fake/sub/dir`,
```

Comment:
It's a little bit hard to understand this test maybe but this line will make the test fail without the patch in the resolve function.